README.md

```diff
@@ -4,88 +4,107 @@
+
+       ```bash
+       # Pyenv <https://github.com/pyenv/pyenv>
+       if [[ -e "${HOME}/.pyenv" ]]
+       then
+           PATH="${HOME}/.pyenv/bin:${PATH}"
+           eval "$(pyenv init -)"
+           eval "$(pyenv virtualenv-init -)"
+       fi
+       ```
+   1. Restart your shell: `exec "$SHELL"`.
+   1. Install the Python version used in this project: `pyenv install`.
+   1. Restart your shell again: `exec "$SHELL"`.
+   1. Verify setup: `diff <(python <<< 'import platform; print(platform.python_version())') .python-version` - should produce no output.
+1. [Install and enable Poetry](https://python-poetry.org/docs/#installation):
+   1. Install:
+
+       ```bash
+       curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+       ```
+   1. Add the following to ~/.bashrc:
+
+       ```bash
+       # Poetry <https://python-poetry.org/>
+       if [[ -e "${HOME}/.poetry" ]]
+       then
+           PATH="${HOME}/.poetry/bin:${PATH}"
+       fi
+       ```
+   1. Restart your shell: `exec "$SHELL"`.
+   1. Verify setup: `poetry --version`.
+1. [Install and enable `nvm`](https://github.com/nvm-sh/nvm#installing-and-updating):
+   1. Install (change the version number if you want to install a different one):
+
+       ```bash
+       curl https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash
+       ```
+   1. Add the following to ~/.bashrc (wraps the upstream instructions to not do anything if `nvm` is not installed):
+
+       ```bash
+       if [[ -d "${HOME}/.nvm" ]]
+       then
+           export NVM_DIR="${HOME}/.nvm"
+           # shellcheck source=/dev/null
+           [[ -s "${NVM_DIR}/nvm.sh" ]] && . "${NVM_DIR}/nvm.sh"
+           # shellcheck source=/dev/null
+           [[ -s "${NVM_DIR}/bash_completion" ]] && . "${NVM_DIR}/bash_completion"
+       fi
+       ```
+   1. Restart your shell: `exec "$SHELL"`.
+   1. Verify setup: `nvm --version`.
+1. [Install latest `npm` LTS](https://github.com/nvm-sh/nvm#long-term-support): `nvm install --lts`
+1. Run `./reset-dev-env.bash` to install packages.
+1. Enable the virtualenv: `. .venv/bin/activate`.
+1. Enable Node.js executables:
+   1. Add the executables directory to your path in ~/.bashrc (replace the project path with the path to this directory):
+
+       ```bash
+       if [[ -d "${HOME}/dev/geospatial-data-lake" ]]
+       then
+           PATH="${HOME}/dev/geospatial-data-lake/node_modules/.bin:${PATH}"
+       fi
+       ```
+   1. Verify setup: `cdk --version`
+
+Re-run `./reset-dev-env.bash` when packages change.
+
+Re-run `. .venv/bin/activate` in each shell.
 
 
 ## AWS Infrastructure Deployment (CDK Stack)
+* Set AWS profile name: `export AWS_PROFILE=<AWS-PROFILE-NAME>`
 * Get AWS credentials (see: https://www.npmjs.com/package/aws-azure-login)
 
     ```bash
-    $ ./node_modules/.bin/aws-azure-login -p <AWS-PROFILE-NAME>
```

Comment:
I think you must specify 'AWS-PROFILE-NAME` otherwise it will probably overwrite your default profile which can create mess in your AWS credentials and can be even dangerous.

Reply:
Can you be a bit more specific? What is it going to overwrite, and in which circumstances? It works for me, and doesn't change `$AWS_PROFILE`:

```
$ echo $AWS_PROFILE 
li-datalake-nonprod
$ aws-azure-login
Logging in with profile 'li-datalake-nonprod'...
Using AWS SAML endpoint https://signin.aws.amazon.com/saml
? Session Duration Hours (up to 12): 12
Assuming role arn:aws:iam::632223577832:role/AccountAdminRole
$ echo $AWS_PROFILE 
li-datalake-nonprod
```

Reply:
Using `-p` parameter you specify which AWS profile in `~/.aws/config` and `~/.aws/credentials` you want to configure. And if it works like I expect it should, by not specifying any profile `-p` you are configuring your default profile in your `~/.aws` configuration.

Reply:
I my case, I have heaps of AWS profiles in my config files.

Reply:
Per [the documentation](https://github.com/sportradar/aws-azure-login#logging-in) and my experience (I haven't set a default profile in `~/.aws/config`) it uses the value of `$AWS_PROFILE`. This is a common pattern: the configuration file has the lowest priority, followed by environment variables, followed by the command line parameter. By specifying the environment variable I don't have to specify the profile to `aws-azure-login` *or* `cdk`, which makes things a bit easier.

Reply:
Yes, if you use `AWS_PROFILE` variable then it is the same as setting `-p`. I always set `-p`, don't use `AWS_PROFILE` variable. I think using `--profile` or `-p` with CLI is more common. But anyway, one of those ways to say which profile to use has to be mentioned.

Reply:
The first entry in the list is

> Set AWS profile name: `export AWS_PROFILE=<AWS-PROFILE-NAME>`

Is that not what you mean?

Reply:
Yes, I think it is more common to use `--profile` parameter then `AWS_PROFILE` env variable. I would use `--profile`.

Reply:
Cheers, done.